### PR TITLE
Fromage

### PR DESCRIPTION
**Pull request by oserikov**
Продолжение PR #523 -- теперь у меня контроль над этой веткой.

### CODE REVIEW

File: services/fromage/server.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    if prompt == "":
+        prompt = ["What is the image?"]
+    text = ""
+    for p in prompt:
+        text += f"Q: {p}\nA:"
+        model_prompt = inp_image + [text]
+        model_outputs = model.generate_for_images_and_texts(
+            model_prompt, num_words=32, ret_scale_factor=RET_SCALE_FACTOR, max_num_rets=0
+        )
+        text += " ".join([s for s in model_outputs if isinstance(s, str)]) + "\n"
```
Comment (oserikov) [inline services/fromage/server.py:48]: dilyararimovna 
i do not clearly understand what are you doing. every time you iterate over the previous utterances in the dialog?

ciwwwnd 
неа, он итерируется по аутпутам, который получает от generate_for_images_and_texts, а так использую только последнее текстовое вхождение

просто generate_for_images_and_tests пикчи тоже может генерить

File: skill_selectors/rule_based_selector/connector.py [hunk 2]
```diff
@@ -97,6 +97,9 @@ async def send(self, payload: Dict, callback: Callable):
             dialog_len = len(dialog["human_utterances"])
             if user_uttr.get("attributes", {}).get("image") is not None:
                 skills_for_uttr.append("dff_image_skill")
+            if any("image" in user_uttr.get("attributes", {}) for user_uttr in dialog["human_utterances"][-5:]):
```
Comment (oserikov) [inline skill_selectors/rule_based_selector/connector.py:100]:
```suggestion
            if any(["image" in user_uttr.get("attributes", {}) for user_uttr in dialog["human_utterances"][-5:]]):
```

File: services/fromage/server.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    raise e
+
+app = Flask(__name__)
+logging.getLogger("werkzeug").setLevel("WARNING")
+
+
+def generate_responses(image_path, prompt):
+    inp_image = [utils.get_image_from_url(image_path)]
+    if prompt == "":
+        prompt = ["What is the image?"]
```
Comment (oserikov) [inline services/fromage/server.py:40]:
```suggestion
    if prompt == "":
        prompt = ["What is the image?"]
    elif isinstance(prompt, str):
        prompt = [prompt]

```

File: services/fromage/server.py (hunk 1 above)
Comment (oserikov) [reply services/fromage/server.py:51]: I'm not sure, addressed with stict check for prompt being a list , see elif before this for loop

**Comment by oserikov** [comment]
@dilyararimovna @smilni could you please take a look on what's going on here?

**Comment by oserikov** [comment]
@dilyararimovna haha check the reversed comparison

https://github.com/ciwwwnd/dream/compare/frmg...deeppavlov:dream:frmg

**Comment by dilyararimovna** [comment]
@oserikov see the previous PR and make fixes according to last comments please

**Pull request merged**